### PR TITLE
Make the error message of changing decimal type the same as Spark's [databricks]

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>
Related to #5196.

## Rationale for this change
We want to throw the same error message as Spark.

## Changes in this PR.
1. Make the error message of changing decimal type be consistent with Spark's. 
- I add some extra compution in `cannotChangeDecimalPrecisionError` to find which value causes the error, so that we could throw the same error message as Spark.
3. Update `update_test_floor_ceil_overflow` to check the error message more precisely.
- So far I only check the exception type (`java.lang.ArithmeticException` before Spark 3.3.0 and `SparkArithmeticException` after 330). The content of error message is not checked because I don't find a good way to get the value of `data_gen`

### CODE REVIEW

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
CI error: https://github.com/NVIDIA/spark-rapids/runs/7067135922?check_suite_focus=true#step:2:5245,
which seems not related to this PR/

**Comment by HaoYang670** [comment]
build

**Review by revans2 — DISMISSED** [review]

File: sql-plugin/src/main/321db/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala [hunk 1]
```diff
@@ -56,4 +59,33 @@ object RapidsErrorUtils {
...
+   * @param toType The type to cast.
+   * @param context The error context, default value is "".
+   */
+  def cannotChangeDecimalPrecisionError(      
+      values: GpuColumnVector,
+      outOfBounds: ColumnVector,
+      fromType: DecimalType,
+      toType: DecimalType,
+      context: String = ""): ArithmeticException = {
+    val row_id = withResource(outOfBounds.copyToHost()) {hcv =>
```
[the remark this answers is not in the document]
Comment (HaoYang670) [reply sql-plugin/src/main/321db/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala:79]: I can extra some common expressions to be a new helper function. But I am not sure where to put the helper.

[the remark this answers is not in the document]
Comment (revans2) [reply sql-plugin/src/main/321db/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala:79]: Only GpuCeil and GpuFloor do this. I would create a trait or an object in mathExpressions.scala and name it something like FloorCeilErrorUtil or RoundingErrorUtil and add it there.

[the remark this answers is not in the document]
Comment (HaoYang670) [reply sql-plugin/src/main/321db/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala]: Updated!

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
The failure seems not related to this PR: https://github.com/NVIDIA/spark-rapids/runs/7124317772?check_suite_focus=true#step:2:34524

**Comment by HaoYang670** [comment]
@revans2 Please have a look. Thank you!

File: sql-plugin/src/main/321db/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala [hunk 2]
```diff
@@ -18,9 +18,9 @@ package org.apache.spark.sql.rapids.shims
 
 import org.apache.spark.sql.catalyst.trees.Origin
 import org.apache.spark.sql.errors.QueryExecutionErrors
-import org.apache.spark.sql.types.DataType
+import org.apache.spark.sql.types.{DataType, Decimal, DecimalType}
 
-object RapidsErrorUtils {
+object RapidsErrorUtils extends {
```
Comment (revans2) [inline sql-plugin/src/main/321db/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala:23]: Why is there an "extends" here that is not used?

**Review by revans2 — APPROVED** [review]
Looks like there is a type in one of the files.

File: sql-plugin/src/main/321db/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala (hunk 2 above)
Comment (HaoYang670) [inline sql-plugin/src/main/321db/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala:23]:
```suggestion
object RapidsErrorUtils {
```

Comment (HaoYang670) [reply sql-plugin/src/main/321db/scala/org/apache/spark/sql/rapids/shims/RapidsErrorUtils.scala]: Updated!

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
The failure test is not related to this PR.

**Review by revans2 — APPROVED** [review]

**Comment by sameerz** [comment]
build

**Pull request merged**